### PR TITLE
Https config with fileDAO record

### PR DESCRIPTION
**Pull request by olhabn**

### CODE REVIEW

File: src/foam/nanos/jetty/HttpServer.js [hunk 1]
```diff
@@ -316,7 +336,7 @@ foam.CLASS({
           server.addConnector(sslConnector);
 
         } catch ( java.io.FileNotFoundException e ) {
-          logger.error("No KeyStore file found at path: " + this.getKeystorePath(),
+          logger.error("No KeyStore file found " + this.getKeyCertificateFileName(),
```
Comment (jlhughes) [inline src/foam/nanos/jetty/HttpServer.js]:
```suggestion
          logger.error("Failed reading keystore", this.getKeyCertificateFileName(), e.getMessage(), 
```

File: src/foam/nanos/jetty/HttpServer.js [hunk 2]
```diff
@@ -325,6 +345,7 @@ foam.CLASS({
           logger.error("Error when enable the https.");
```
Comment (jlhughes) [inline src/foam/nanos/jetty/HttpServer.js]:
```suggestion
          logger.error("Failed configuring https", e);
```

**Review by jlhughes — APPROVED** [review]

File: src/foam/nanos/jetty/HttpServer.js [hunk 3]
```diff
@@ -99,10 +107,15 @@ foam.CLASS({
       class: 'FObjectProperty',
       of: 'foam.nanos.logger.Logger',
       visibility: 'HIDDEN',
-      javaFactory: `
+      javaGetter: `
```
Comment (olhabn) [inline src/foam/nanos/jetty/HttpServer.js]: NPE exception line 138:
ok, so getX().get("logger") is not null when we call javaFactory but:
in DAOLogger's delegate is null although it shouldn't be and it isn't in other places. It is localLogMessageDAO.

so logger == null check will not catch it it. javaGetter fixes it but it doesn't feel right.

Comment (jlhughes) [reply src/foam/nanos/jetty/HttpServer.js]: Ok, just explicitly use a StdoutLogger from here, and we'll look into setup order later.

Comment (olhabn) [reply src/foam/nanos/jetty/HttpServer.js]: the problem occur after http server record updated.

maybe in some put methods context there is no localLogMessageDAO?

**Pull request merged**